### PR TITLE
introduce sidecar feature gate

### PR DESCRIPTION
**Pull request by mfranczy**
**What this PR does / why we need it**:
It introduces a sidecar feature gate in order to give cluster administrator some control on executed code within pod's environment

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
- [x] add functional and unit tests
- [x] update user guide

**Release note**:

```release-note
introduce sidecar feature gate
```

### CODE REVIEW

**Comment by mfranczy** [comment]
https://github.com/kubevirt/user-guide/pull/263 updated user guide

**Comment by davidvossel** [comment]
```
Causes: [
                    {
                        Type: "FieldValueInvalid",
                        Message: "sidecar feature gate is not enabled in kubevirt-config, invalid entry metadata.annotations.hooks.kubevirt.io/hookSidecars",
                        Field: "metadata.annotations",
                    },
                ],
```

The code structure looks great, but there's something wrong with the functional tests.

**Comment by mfranczy** [comment]
@davidvossel thanks for the review, tests failed because I forgot to update testing manifest. Let's wait for CI and see how it goes now.

**Comment by slintes** [comment]
let's wait for the test lanes which did not run yet on the latest commit, but I guess this good to be merged

/lgtm
/approve

**Comment by mfranczy** [comment]
@slintes thanks for the review, I got the appropriate polarion id so I had to update the PR.
@jerry7z please check if it's ok.

File: tests/vmi_hook_sidecar_test.go [hunk 1]
```diff
@@ -52,7 +55,8 @@ var _ = Describe("HookSidecars", func() {
 		vmi.ObjectMeta.Annotations = RenderSidecar(hooksv1alpha1.Version)
 	})
 
-	Describe("VMI definition", func() {
+	Describe("[rfe_id:2666][crit:medium][vendor:cnv-qe@redhat.com][level:component] VMI definition", func() {
```
Comment (jerry7z) [inline tests/vmi_hook_sidecar_test.go]: rfe id is 2667

File: tests/vmi_hook_sidecar_test.go [hunk 2]
```diff
@@ -96,8 +100,24 @@ var _ = Describe("HookSidecars", func() {
...
+		Context("with sidecar feature gate disabled", func() {
+			BeforeEach(func() {
+				tests.DisableFeatureGate(virtconfig.SidecarGate)
+			})
+
+			AfterEach(func() {
+				tests.EnableFeatureGate(virtconfig.SidecarGate)
+			})
+
+			It("[test_id:2667]should not start with hook sidecar annotation", func() {
```
Comment (jerry7z) [inline tests/vmi_hook_sidecar_test.go]: test case id is 2666

File: tests/vmi_hook_sidecar_test.go (hunk 1 above)
Comment (mfranczy) [reply tests/vmi_hook_sidecar_test.go]: fixed

File: tests/vmi_hook_sidecar_test.go (hunk 2 above)
Comment (mfranczy) [reply tests/vmi_hook_sidecar_test.go]: fixed

**Comment by mfranczy** [comment]
@slintes all comments have been addressed, let's wait for CI and after that, I think we can merge it.

**Pull request merged**